### PR TITLE
Adds styleguide-ui package

### PR DESCRIPTION
**Pull request by jonsamp**
# Why

We'd like to start moving UI components into our styleguide UI package. This PR begins adding those components

# Changes

- Adds package/styleguide-ui, which contains
  - Scripts to fetch our icon set from Figma, then makes React components of them all (>2,000 icons)
  - A `<ButtonBase>` component. This is a proof of concept to show that we can export design systems components.
    - *To use these components, styleguide-ui must be in a website that has styleguide set up, since styleguide-ui relies on tailwind classes + styles from that package.

- This package now properly tree shakes components. Before, when importing any one component from @expo/styleguide, we would download them all. With this change, we can expect to see about 220Kb less download on all pages once we migrate to styleguide-ui.

<details>
<summary>
Bundle analysis before
</summary>
before
</details>
<details>
<summary>
Bundle analysis after
</summary>
after
</details>

# Screenshots
Screenshot 2023-02-03 at 2 51 38 AM
Screenshot 2023-02-03 at 2 51 46 AM 1

# Test plan

Read the **README.md** in /styleguide-ui to learn how to get the credentials you need to generate the icons. Then run `yarn build` in the /styleguide-ui (takes about 2 minutes to pull all icons + build them all). Then run `yarn dev` in the root of styleguide. Once finished, visit localhost:3000 to preview the icons and the button base component.

Also `yarn link` this new package in a new project with webpack bundle analyzer and make sure that when you import components from /stylguide-ui, you only get the ones you import (aka, make sure that tree shaking is working correctly).

# Question

Should this be a separate package from @expo/styleguide?
- styleguide-ui relies on styles from styleguide
- I can't see a use case where we'd practically use them separately

If we want all components to be in @expo/styleguide. then I can refactor this PR to combine all the new logic + add in all the other things styleguide exports, like colors, etc.

### CODE REVIEW

File: packages/styleguide-ui/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+## Get started
+
+1. Install dependencies with `yarn`.
+2. Build everything with `yarn build`.
+
+### Icons
+
+We generate all icon files based on our Figma icons. The process is:
+
+1. Make a call to Figma to get all the icons from a specific doc. To do this, you'll need to:
```
Comment (tchayen) [inline packages/styleguide-ui/README.md:12]: When I followed those instructions for the first time it wasn't clear to me how to make the Figma call until I retried `yarn build`. I'd suggest some re-ordering here.

Maybe the `### Icons` section could become some `### Prerequisites` at the top of `## Get started`, and then in the line with `yarn build` we could add a note that this step will fetch the icons?

**Review by tchayen — COMMENTED** [review]
Wow. I didn't know such thing with downloading icons from Figma is already possible 🤩. I have one thought that came to me when I was setting up the project.

File: packages/styleguide-ui/package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    "@types/react": "^18.0.27",
+    "cheerio": "^1.0.0-rc.12",
+    "dotenv": "^16.0.3",
+    "npm-run-all": "^4.1.5",
+    "react": "^18.2.0",
+    "rimraf": "^4.1.2",
+    "rollup": "^3.12.1",
+    "tslib": "^2.5.0",
+    "typescript": "^4.9.5"
+  },
```
Comment (Simek) [inline packages/styleguide-ui/package.json:47]: I can see few unused packages listed in there (unless they are peers of something), like `cherio` and `tslib`.

**Comment by FiberJW** [comment]
this is really cool! I second Tomek and Simek's suggestions around improving the readme and removing any potentially unused dependencies

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,2 @@
 .DS_Store
-.idea
```
Comment (Simek) [inline .gitignore:2]: Can we leave IDEA in?

**Comment by jonsamp** [comment]
Closing in favor of #46

**Pull request closed without merging**